### DISCUSSION TITLE
Best way to do multiple logos in a RevealJS slide deck?

### DISCUSSION THREAD

**Discussion by MattF-NSIDC** [quarto-dev/quarto-cli#7668] [Q&A] [answered]
### Description

I've had no issues setting up a single logo for slide decks, but I'm working on a presentation now for AGU where I really should have around 5 logos on each slide to recognize funding sources.

I can do this with HTML tags in the footer, or I could make one combined image for all the logos. Are there better ways to go about this? Could Quarto one day accept an array of logos in config?

Thanks!

**Comment by mcanouil** [comment]
- Use footer
- Use CSS classes and absolute positioning, *e.g., on the title slide.
- Use several or one combined image
- ...
- Make a funding/acknowledgment slide
- Define your own Reveal.js template and/or redefine the title slide partial.

It is really up to you. The "better way" is the way that works for you.

> Could Quarto one day accept an array of logos in config?

I would say unlikely because that's not the purpose of `logo`.
What you want here is sponsorship/fundings which is different from logo.

**Reply by MattF-NSIDC** [reply]
>It is really up to you. The "better way" is the way that works for you.

I guess I should have been more clear about the requirements from my institution, and that this is a really common requirement: I need to show all multiple logos on every slide. My personal preference is to maximize maintainability, which is why I'm leaning towards some HTML in a `footer` over a new reveal.js template.

> that's not the purpose of logo.  
> What you want here is sponsorship/fundings which is different from logo.

What is the purpose of `logo`? What I'm trying to do is match our powerpoint templates with required logos on every slide, like so, and from a naive perspective without knowledge of such design decisions, this looks really close to what I perceive(d) as the intended purpose of `logo` in Quarto and I intuited I might be able to just pass in an array of images to `logo`:

![image](https://github.com/quarto-dev/quarto-cli/assets/18539827/6e709e6c-c719-4a1a-8fb5-ae2a5285ca65)

They look like and are positioned in the same way as `logo` in Quarto, except there are multiple!

This is more about "affiliation" than about "sponsorship" or "funding" (poor choice of words in OP) -- our team did not receive funding from some of the institutions I must include; in fact, some of them are taking overhead funding _from_ the project grant :) When presenting, there's pressure (sometimes a requirement, depending on venue) to recognize all affiliated institutions, and in my institution, that looks like including all affiliated logos on every slide.

I think this need is common in research institutions, and I hope the Quarto team will reconsider a way to make multiple logos achievable as a first-class configuration option!

**Reply by mfisher87** [reply]
>Define your own Reveal.js template and/or redefine the title slide partial.

Looking through the docs and I've seen this section on "custom template", but it only describes the latter half of what you suggested; redefining the title slide partial. Is there documentation on overriding the full revealJS template? Or overriding the template for individual slides?

https://quarto.org/docs/presentations/revealjs/advanced.html#custom-template

I'd be happy to open a PR to update this doc page to include this information.

This would be a great solution for my institution because it would make it easier to re-use. But often projects need to add their own optional logos, and extending the set of logos would be harder with the revealJS template than the powerpoint template. Everything is a trade-off I suppose! I'm still interested in exploring the template option, though.

EDIT: To clarify, I see [the `template` option](https://quarto.org/docs/reference/formats/presentations/revealjs.html#rendering) but I don't have context about what goes in it, like an example. I'm guessing I can find an example in the pandoc source, so starting to dig there.

**Reply by mfisher87** [reply]
After digging in pandoc source, I found `default.revealjs` template, which didn't help much because it also has a `$body$` variable where the non-title slides should go and I'm not seeing docs about how it's populated. I'm not Haskell-literate, so I'm not having much luck in the source code either... I'm not sure where to go from here to customize the revealJS template (aside from the documented title partial, of course)!

**Reply by mcanouil** [reply, marked as answer]
Look at the extension part of the documentation (and/or https://github.com/mcanouil/awesome-quarto#presentations), you'll see several revealjs template you can get inspiration from.
Then you try to make your own, which could be exactly how you need it to be.

**Reply by mfisher87** [reply]
Nice! Thanks for creating `awesome-quarto` :)
